src/libstore/derivations.hh

```diff
@@ -12,11 +12,24 @@ namespace nix {
 
 /* Abstract syntax of derivations. */
 
+/// Pair of a hash, and how the file system was ingested
+struct FileSystemHash {
+    FileIngestionMethod method;
+    Hash hash;
+    FileSystemHash(FileIngestionMethod method, Hash hash)
+        : method(std::move(method))
+        , hash(std::move(hash))
+    { }
```

Comment:
We can remove this constructor (and the others) and use designated initializers (`DerivationOutputHash { .method = ..., .hash = ... }`).